app/controllers/checkout_controller.rb

```diff
@@ -150,10 +161,13 @@ def before_payment
   def valid_payment_intent_provided?
     return false unless params["payment_intent"]&.starts_with?("pi_")
 
-    last_payment = OrderPaymentFinder.new(@order).last_payment
-    @order.state == "payment" &&
-      last_payment&.state == "requires_authorization" &&
-      last_payment&.response_code == params["payment_intent"]
+    return false unless @payment = Spree::Payment.where(
+      response_code: params["payment_intent"],
+      state: "requires_authorization"
+    ).first
+
+    @order ||= @payment.order
+    @order.state == "payment" && @payment.order == @order
```

Comment:
The second half of this line will fail / return false if the payment's order isn't the order that's currently loaded in the user's browser session, right? And then it won't be processed? :thinking:

Reply:
I think that snuck in there by mistake, I was trying out and squashing a bunch of different things. I think it's better to process the payment intent regardless of the current order actually, I'll push a change.